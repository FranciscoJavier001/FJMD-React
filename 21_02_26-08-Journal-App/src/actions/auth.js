//** Estas son acciones que vamos a definir */
import { firebase, googleAuthProvider } from '../firebase/firebase-config'
import { types } from '../types/types'

//** Esto es una tarea asincrona, osea que se va a ejecutar todo el proceso, luego se va a encontrar con el dispatch que ejecuta la accion del login que es la que se va a aplicar directamente en el store (la fuente unica de la verdad), que modifica el state y nos regresa todo lo que estamos esperando  */
export const startLoginEmailPassword = (email, password) => { //** Esta funcion va a recibir el email y el password */
    return (dispatch) => { //** Esta funcion va a regresar un callback, osea una funcion a disparar */
        //** Hay que llamar a otro dispatch cuando ya se ejecute todo, cuando ya tenga la data */
        setTimeout(() => {
            dispatch( login( 123, 'Pedro')) //** Aqui voy a hacer un dispatch del login, con el usuario */
        }, 3500);
    }
}

export const startRegisterWithEmailPasswordName = ( email, password, name ) => { //** Esta es una tarea asincrona, asi que voy a tener un callback */
    return ( dispatch ) => { //** Cuando tenga el usuario grabado en firebase voy a hacer el dispatch */
        firebase.auth().createUserWithEmailAndPassword( email, password ) //** Aqui mandamos los parametros de email y password */
        .then( async ({ user }) => { //** Esta va a ser una promesa */

            await user.updateProfile({ displayName: name }) //** que me de el name que estoy recibiendo como argumento de mi funcion y esto me regresa una promesa */

            dispatch( //** Esto es para llamar de nuevo el dispatch */
                login( user.uid, user.displayName)
            )
        })
        .catch( e => { //** Esto es para atrapar el error y mostrarlo */
            console.log(e);
        })
    }
}

export const startGoogleLogin = () => { //** Una funcion de fleca que no recibe nada, pero es una tarea sincrona voy a poner el return y el callback me va a proveer el dispatch, por lo cual cuando tenga la informacion lo voy a llamar */
    return ( dispatch ) => {
    firebase.auth().signInWithPopup( googleAuthProvider ) //** Esto va a retormarme una promesa */
        .then( ({ user }) => { //** Aqui voy a usar la desestructuracion para extraer el user */
            dispatch( //** Que cuando esto se responda voy a hacer el dispatch de la accion que ya tengo definido abajo con el login, donde voy a poder el id que viene del user.uid, y el displayName que viene del user.displayName */
                login( user.uid, user.displayName)
            )
        })
    }
}

export const login = (uid, displayName) => ({ //** Esta es la accion que voy a llamar cuando tenga el uid y el displayNameny esta accion me debe regresar un type.login */
    type: types.login,
    payload: { //** El payload va a ser un objeto que tenga un uid y el displayName */
        uid,
        displayName
    }
})