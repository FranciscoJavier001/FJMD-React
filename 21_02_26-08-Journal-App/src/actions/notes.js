import { db } from "../firebase/firebase-config";
import { types } from '../types/types'

//** Voy a exportar un nuevo componente */
export const startNewNote = () => { //** Esta es una tarea asyncrona, asi que voy a poner el return con un callback que va a ocupar el dispatch, y lo voy a llamar en el dispatch del Sidebar.js en el handleAddNew */
    return async( dispatch, getState ) => { //** Aqui voy a poner el segundo argumento, despues del dispatch que vamos a llamar getState (pero ese segundo argumento es una funcion para obtener el state similar al useSelector) */
        //** Para grabar en Firestore voy a oupar otras cosas como el ID del usuario */
        const uid = getState().auth.uid //** Este es todo el State */
        console.log( uid );

        //** Ahora voy a crear la nueva nota que quiero guardar */
        const newNote = {
            title: '', //** Este lo voy a inicializar vacio */
            body: '',
            date: new Date().getTime() //** Este me va a dar el tiempo exacto cuando la nota fue creada, es un objeto de JS, y lo voy a mandar a un path de Firestore para guardarlo */
        }

        //** Para hacer lo del getTime, debo de conseguir la referencia a la base de datos, esta ya la tengo en firebase & firebase-config.js */
        const doc = await db.collection(`${ uid }/journal/notes`).add( newNote ) //** Voy a hacer la referencia al documento y esto viene de db que lo explique arriba, esto lo voy a poner en la coleccion donde quiero guardar esto, voy a ocupar el uid del usuario , luego el journal que va a ser donde lo estoy guardando, luego notes, porque quiero que ahi este y ahi ya puedo hacer el .add y ahi mando el newNote, el add regresa una promesa que retorna el document.reference y como esto es una promesa en el return del startNewNote lo voy a convertir a async y aqui hacer el await, osea espera que se haga la insercion y el doc va a dar referencia al documento */

        //** Aqui voy a mandar llamar la accion que defini abajo */
        dispatch( activeNote( doc.id, newNote ) ) //** Esta pide el id de la nota y este lo tengo en el doc.id, y luego viene la nota que seria el newNote que lo acabo de crear */
    }
}

//** Lo ultimo despues de todo lo que hemos echo es manejar esta accion el noteReducer */
//** Voy a crear una nueva accion que sea syncrona porque ya voy a tener la informacion voy a mandar algo al reducer  */
export const activeNote = ( id, note ) => ({ //** Este va a recibir el id y la nota */
    //** Este objeto va a tener los types */
    type: types.notesActive,
    payload: {
        id,
        ...note
    }
})

//** Como voy a guardar las notas en el store voy a guardar esto, con esto estoy creando siempre un nuevo arreglo */
export const setNotes = ( notes ) => ({ //** Aqui voy a recibir las notas */
    type: types.notesLoad, //** el type que va a tener esto , va a ser de notesLoad y el paylod va a ser todas las notas que este recibiendo como argumento */
    payload: [ notes ]
})