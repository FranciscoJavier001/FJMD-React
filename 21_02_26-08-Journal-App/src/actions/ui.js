//** Acciones en el UI */

import { types } from "../types/types";

export const setError = ( err ) => ({ //** De una vez voy a hacer el return del objeto y va a recibir el payload del error */
    type: types.uiSetError,
    payload: err
})

export const removeError = () => ({ //** De una vez voy a hacer el return del objeto y no va a recibir ningun objeto */
    type: types.uiRemoveError, //** Va a llamar a la funcion de remover error */
})

export const startLoading = () => ({ //** Asi llamamos las acciones, y ahora se debe hacer el dispatch de estas acciones */
    type: types.uiStartLoading,
})

export const finishLoading = () => ({
    type: types.uiFinishLoading,
})