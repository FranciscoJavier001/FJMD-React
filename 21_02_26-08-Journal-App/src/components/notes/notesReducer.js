//** Aqui voy a poner como quiero que este el State */

import { types } from "../../types/types";

//** Va a tener notes como un arreglo, ese es el primero y el segundo va a ser el active, que puede ser null o va a poder tener el objeto de la nota activa, si tenemos null, no hay ninguna nota seleccionada y si esta activa, va a tener el id, que va a ser un id unico de firebase y el title como un string y el body que tambien va a ser un string , la url de la imagen y una fecha  */

//** El reducer va a ser una funcion pura */
const initialState = { //** Este va a ser el estado inicial del reducer */
    notes: [], //** El objeto inicial va a ser notes como un objeto vacio */
    active: null
}

export const notesReducer = ( state = initialState, action ) => { //** Recuerda primero ponemos lo que recibe, y el state va a ser igual al initialState */

    switch (action.type) { //** Aqui voy a empezar a definir muchos tipos */

        //** Despues de hacer lo de la nueva nota con el dispatch en el notes, voy a poner el tipo aqio */
        case types.notesActive:
            return { //** Aqui lo que voy a hacer va a ser regresar un nuevo estado, osea la clonacion del estado anterior, para siempre regresar un nuevo estado, no mutar el anterior */
                ...state,
                active: { //** Voy a regresar la nota activa que va a ser un objeto del action.payload */
                    ...action.payload
                }
            }

        case types.notesAddNew:
            return {
                ...state,
                notes: [ action.payload, ...state.notes ]
            }

            //** Voy a editar el reducer para saber que hacer cuando recibamos una tarea o una accion de este tipo */
            case types.notesLoad: //** Y que es lo que voy a hacer, voy a retornar un nuevo estado con el operador spred del state y las notas van a ser igual al action.payload, pero esto es un arreglo asi que voy a exparsirlo con el operador spred */
            return {
                ...state,
                notes: [ ...action.payload ]
            }

            //** Debo de implementar lo que va a hacer la accion de refreshNote */
            case types.notesUpdated:
                return {
                    ...state, //** Siempre hacer el spred en los types del estado anterior para la modificacion o cualquier cosa */
                    notes: state.notes.map( //** Ahora voy a tomar las notas que haya con el state y el map regresa un nuevo arreglo y voy a mutar la nota utilizando un operador ternario, voy a evaluar si la nota es igual al action.payload.id, que sea el id que estoy mandando de notes, si son iguales significa que es la nota que necesito actualizar entonces si el id es el mismo voy a retornar el action.payload.note, que el note es el note del payload de notes, caso contrario significa que no hay ninguna modificacion y pongo la note asi como esta */
                        note => note.id === action.payload.id
                        ? action.payload.note
                        : note
                    )
                }

                case types.notesDelete:
                    return {
                        ...state, //** Esparzamos el state para no perder el estado anterior */
                        active: null,
                        notes: state.notes.filter( note => note.id !== action.payload ) //** Esto quiere decir que debe regresar todas, menos la que yo seleccione */
                    }

                case types.notesLogoutCleaning:
                    return {
                        ...state, //** Voy a hacer el clon del State */
                        active: null, //** La nota activa va a ser igual a null */
                        notes: [] //** Va a ser igual a un arreglo vacio */
                    }

        default:
            return state; //** Solamente voy a hacer el return del state */
    }
}