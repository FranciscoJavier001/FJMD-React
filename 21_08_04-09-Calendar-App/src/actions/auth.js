//** ________________________________________________________________________________________________________________________________________________ */
import { fetchConToken, fetchSinToken } from "../helpers/fetch"
import { types } from "../types/types"
import Swal from 'sweetalert2' //** Para las alertas */

//** Funcion que comienza proceso de autentificacion */
export const startLogin = ( email, password ) => { //** Exporto(mando) la funcion que va a ser asyncrona, recibe email, password */
    return async( dispatch ) => { //** Por el thunk se va a volver a disparar */
        
        // console.log(email, password); //** Para mostrarlo en consola pero desde aqui */

        //** endpoint=auth, recibe la data(desestructurada), y luego el metodo que es POST */
        const resp = await fetchSinToken( 'auth', { email, password }, 'POST' ) 
        const body = await resp.json() //** Es igual a los parametros del resp=(name/ok/token/uid) como un json(es el msg/ok) */

        // console.log(body);

        if( body.ok ) { //** Si tenemos el ok, osea para saber si hizo la conexion */
            localStorage.setItem( 'token', body.token ) //** Voy a guardar en localStorage el token que lo tengo en el body */
            localStorage.setItem( 'token-init-date', new Date().getTime() ) //** token-init-date para saber la fecha de cuando se creo */

            dispatch( login({ //** Hago el dispatch de la info Consola>Redux>state>auth linea 29 */
                uid: body.uid, //** Mando el login y recibo un objeto con el uid y el name que vienen en el body */
                name: body.name //** Establece el uid y el name en el store */
            }) )
        } else { //** Si escribe mal la contraseÃ±a o el usuario */
            Swal.fire('Error', body.msg, 'error') //** Primero lo que dice, segundo el msg que tengo en la base de datos, tercero tachita */
        }
    }
}

//** Funcion starRegister, solo necesitamos name, email, password */
export const startRegister = ( email, password, name ) => { //** Esta funcion la exporte en components>auth>LoginScreen */
    return async( dispatch ) => { //** Funcion Asyncrona, y el dispatch para poder disparar alguna accion, cuando tenga el usuario */

        const resp = await fetchSinToken( 'auth/new', { email, password, name }, 'POST' ) //** endPoint, variables que mandamos y methodo */
        const body = await resp.json() //** Respuesta se manda por el body>raw,>JSON, porque es el objeto con los parametros, name/ok/token/uid) */

        if( body.ok ) { //** Si el body esta entonces */
            localStorage.setItem( 'token', body.token ) //** Guardamos lo que manda en el body.token */
            localStorage.setItem( 'token-init-date', new Date().getTime() ) //** Guardamos el valor de creacion de a variable */

            dispatch( login({ //** Hacemos el dispatch del login */
                uid: body.uid, //** Estos son los parametros del login */
                name: body.name
            }) )
        } else { //** Sino tenemos el error */
            Swal.fire('Error', body.msg, 'error')
        }
    }
}


export const startChecking = () => { //** No requiere argumentos */
    return async( dispatch ) => { //** Retorno el dispatch y va a ser una tarea async */

        const resp = await fetchConToken( 'auth/renew' ) //** endPoint, no necesita ningun argumento y es un GET */
        const body = await resp.json() //** Respuesta se manda al body>raw,>JSON, porque en json tengo nuevo token */

        // console.log(body); //** Para ver la impresion en consola del body, para ver si mandamos bien los parametros */

        if( body.ok ) { //** Si el body tiene el ok entonces */
            localStorage.setItem( 'token', body.token ) //** Guardamos lo que manda en el body.token, en localStorage variable body */
            localStorage.setItem( 'token-init-date', new Date().getTime() ) //** Guardamos el valor de creacion de a variable (tiempo) */

            dispatch( login({ //** Hacemos el dispatch del login */
                uid: body.uid, //** Estos son los parametros del login, que establecen la informacion */
                name: body.name
            }) )
        } else { //** Sino tenemos el error */
            dispatch( checkingFinish() ) //** De types>types, accion para confirmar que se hizo la autentificacion si/no, de linea 76 */
        }
    }
}

const checkingFinish = () => ({ type: types.authCheckingFinish }) //** Constante que no recibe nada y retorna un objeto de types, la llamo linea 71 */

//** Accion sincrona, y solo la voy a usar aqui */
const login = ( user ) => ({ //** Recibo el user */
    type: types.authLogin, //** Viene de types>types que es para saber que ya se logeo el usuario */
    payload: user //** Payload es lo que se carga del usuario y si no hay nada puede omitirlo */  
})