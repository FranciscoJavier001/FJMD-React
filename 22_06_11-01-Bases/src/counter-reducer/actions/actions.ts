//** Type-Similar Interface */
//** Exporta para poderlo usar fuera de este archivo */
export type CounterAction = //** Voy a definir los tipos de acciones, y el payload va a ser del tipo de valor de la accion */
| { type: 'increaseBy', payload: { value: number } }
| { type: 'decreaseBy', payload: { value: number } } 
| { type: 'reset' }

//** Esto es un Action Creator */
export const doReset = ():CounterAction => ({ 
    type: 'reset' //** Me esta regresando este objeto */
})

export const doIncreaseBy = (value: number):CounterAction => ({ //** En estos casos no olvidar el payload */
    type: 'increaseBy',
    payload: { value } //** Lo mando porque estoy reibiendo un value */
})

export const doDecreaseBy = (value: number):CounterAction => ({
    type: 'decreaseBy',
    payload: { value }
})