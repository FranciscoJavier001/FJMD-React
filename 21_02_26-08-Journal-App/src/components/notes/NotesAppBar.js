import React from 'react'
import { useDispatch, useSelector } from 'react-redux'
import { startSaveNote } from '../../actions/notes'

export const NotesAppBar = () => {

    const dispatch = useDispatch()
    const { active } = useSelector( state => state.notes ) //** Voy a extraer la nota activa, aunque ya la tenga del getState en notes */
    
    const handleSave = () => { //** Voy a hacer el dispatch de la accion, pero voy a importar de rediux el useDispatch */
        dispatch( startSaveNote( active ) )//** Despues de importarlo ya puedo hacer el dispatch del StartSaveNote */
    }

    return (
        <div className="notes__appbar">
            <span>14 de Marzo del 2021</span>

            <div>
                <button className="btn"> {/* En bootstrap implementamos un boton */}
                    Picture
                </button>

                <button
                    className="btn" /* En bootstrap implementamos un boton */
                    onClick={ handleSave }//** Voy a hacer el dispatch de la accion, osea que haccion va a hacer cuando le demos click, voy a crearme la funcion */
                >
                    Save
                </button>
            </div>
        </div>
    )
}