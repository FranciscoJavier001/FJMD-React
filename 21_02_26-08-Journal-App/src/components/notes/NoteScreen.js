import React from 'react'
import { useSelector } from 'react-redux'
import { useForm } from '../../hooks/useForm'

import { NotesAppBar } from './NotesAppBar'

export const NoteScreen = () => {

    //** Aqui voy a manejar el manejo de los cambios, este es un componente que va a ser un poco mas complejo, porque tiene mucha logica enuelta aqui */
    //** Primero necesito la referencia a a la nota activa, para eso voy a utilizar el useSelector y para eso voy a seleccionar los notes, asi obtengo la referencia pero solo me interesa la nota activa */
    const { active:note } = useSelector(state => state.notes ) //** Con los : renombro como quiero ponerle */
    //** Aqui voy a utilizar el useForm y este hook funciona haciendo la desestructuracion de lo que me va a regresar el useForm me regresa mis formValues y tambien el handleInputChange y el useForm recie los campos que debe tener los campos del formulario y le puedo mandar la note que tengo arriba eso haria que yo tenga la informacion en el formValues y de ahi ya la puedo extraer */
    const [ formValues, handleInputChange ] = useForm( note )

    //** Ahora voy a extraer el body y el title que viene en el formValues */
    console.log(formValues);
    const{  body, title } = formValues; //** Voy a tomar el body y el title y lo voy a poner en sus respectivas cajas */
    

    return (
        <div className="notes__main-content">

            <NotesAppBar /> {/* Igual no va a recibir ningun argumento */}

            <div className="notes__content"> {/* Este va a ser el agrupador de mi formulario */}

                <input
                    type="text"
                    placeholder="Some Awesome Title"
                    className="notes__title-input"
                    autoComplete="off"
                    value={ title } //** Este valor lo extraemos con el formValues */
                    onChange={ handleInputChange }//** La funcion para manejar su cambio va a ser onChange igual al handleInputChange */
                />

                <textarea
                    placeholder="What Happend Today"
                    className="notes__textarea"
                    value={ body } //** Este valor igual lo extraemos con el formValues */
                    onChange={ handleInputChange }
                ></textarea>

                {/* La imagen la voy a mostrar de manera condicional */}
                {
                    //** Le vamos a decir si en la note.url, entonces voy a mostrar este componente */
                    (note.url) 
                    && (
                        <div className="notes__image">
                            <img
                            src="https://image.freepik.com/foto-gratis/silueta-mujer-joven-meditando-practicando-yoga-playa-al-atardecer_35708-291.jpg"
                            alt="imagen"
                            />
                        </div>
                    )
                }

            </div>
            
        </div>
    )
}
