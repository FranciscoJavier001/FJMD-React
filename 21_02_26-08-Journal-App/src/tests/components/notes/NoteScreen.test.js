//** Esto lo copie de AppRouter.test.js */
import React from 'react' //** Por que voy a utilizar JSX */
import { mount } from 'enzyme' //** Recuerda para usar el mount */
import { Provider } from 'react-redux'

import configureStore from 'redux-mock-store' //ES6 modules
import thunk from 'redux-thunk'

import '@testing-library/jest-dom'
import { activeNote } from '../../../actions/notes'
import { NoteScreen } from '../../../components/notes/NoteScreen'

//** Aqui voy a fingir el activeNote */
jest.mock('../../../actions/notes', () => ({
    activeNote: jest.fn(),
}))

/** Crea los middleware, configura el mockStore */
const middlewares = [thunk] //** thunk es el middleware */
const mockStore = configureStore(middlewares)

const initState = { //** Este contiene el estado del store en este momento, pero lo vamos a desestructurar */
    auth: { //** Lo voy a tener como un objeto con los siguientes parametros */
        uid: '1',
        name: 'Frank'
    }, 
    ui: { //** Y este es como lo voy a tener en el estado inicial del test */
        loading: false,
        msgError: null
    },
    notes: { //** La activeNote(nota activa) la voy a dejar en null */
        active: { //** Con la nota activa voy a poner un objeto que tenga un id(aleatorio), el tittle, body, date (no importa mucho) */
            id: 1234,
            title: 'Hola',
            body: 'Mundo',
            date: 0
        }, 
        notes: [] //** La nota la voy a dejar vacia */
    }
}

//** Voy a crear un store que es un objeto */
let store = mockStore(initState)
store.dispatch = jest.fn() //** Asi vamos a evaluar como son llamadas las funciones, osea vamos a reemplazar la funcion que tenga el store por una funcion jest la cual si voy a poder evaluar y voy a tener control absoluto sobre la funcion */

const wrapper = mount( //** Voy a ocupar renderizar mas cosas */
    <Provider store={ store }> {/* Aqui voy a proveer el store que va a ser igual sl store del mockStore que se creo en la arriba en la linea 18 */}
        <NoteScreen /> 
    </Provider>
)

describe('Pruebas en <NoteScreen />', () => {
    
    test('debe de mostrarse correctamente', () => {
        
        expect(wrapper).toMatchSnapshot()
    })

    //** Cuando cabia algo en la caja de texto, se usa el efecto y cambia el activeNote, osea para hacer la prueba voy a modificar una caja de texto y esto se debe de disparar con esos argumentos */
    test('debe de disparar el active note', () => {
        
        wrapper.find('input[name="title"]').simulate('change', { //** Como lo tengo en el textArea puedo especificarlo entre llaves cuadradas y entre doble comillas la que quiera, esto esta en la linea 59 del notesScreen, voy a llamarlo con la simulaion change, le pongo "," y este tiene su target y este tiene el name que va a ser el title y el title que puede ser lo que sea, con esto hago el cambio en la caja de texto */
            target: {
                name: 'title',
                value: 'Hola de nuevo'
            }
        })

        //** Ahora me toca ver si fue llamado */
        expect( activeNote ).toHaveBeenLastCalledWith(
            1234, //** Un id Cualquiera */
            { //** Un id que tenga el body, title, id, date, estos de manera aleatoria */
                body: 'Mundo',
                title: 'Hola de nuevo',
                id: 1234,
                date: 0
            }
        )
    })
})
