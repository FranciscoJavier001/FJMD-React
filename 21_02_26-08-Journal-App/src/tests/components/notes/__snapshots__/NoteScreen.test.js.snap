// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <NoteScreen /> debe de mostrarse correctamente 1`] = `
Array [
  " ",
  <div
    className="notes__main-content"
  >
    <div
      className="notes__appbar"
    >
      <span>
        14 de Marzo del 2021
      </span>
      <input
        id="fileSelector"
        name="file"
        onChange={[Function]}
        style={
          Object {
            "display": "none",
          }
        }
        type="file"
      />
      <div>
        <button
          className="btn"
          onClick={[Function]}
        >
          Picture
        </button>
        <button
          className="btn"
          onClick={[Function]}
        >
          Save
        </button>
      </div>
    </div>
     
    <div
      className="notes__content"
    >
       
      <input
        autoComplete="off"
        className="notes__title-input"
        name="title"
        onChange={[Function]}
        placeholder="Some Awesome Title"
        type="text"
        value="Hola"
      />
      <textarea
        className="notes__textarea"
        name="body"
        onChange={[Function]}
        placeholder="What Happend Today"
        value="Mundo"
      />
    </div>
    <button
      className="btn btn-danger"
      onClick={[Function]}
    >
      Delete
    </button>
  </div>,
]
`;
